### ISSUE TITLE
-1 as an invalid mob type

### ISSUE THREAD

**Issue by Camotoy** [GeyserMC/MCProtocolLib#539]
Thanks for pushing a (potential) fix for the invalid sound mapping. Over at Geyser we've gotten a report of an invalid mob type: https://github.com/GeyserMC/Geyser/issues/502. Easy fix but I'm not sure what -1 should default to.

**Comment by Steveice10** [comment]
A workaround should be in now, let me know if it solves the problem. Honestly, the proper solution would be to go through adding all the defaults that Minecraft uses in it's registries, but I'm worried that implementing catch-all defaults would end up hiding real errors in value mappings.

**Comment by Steveice10** [comment]
Closing as fixed and tracking implementation of a proper fix in #540.

**Comment by Camotoy** [comment]
Awesome, thanks.

**Comment by Steveice10** [comment]
As an update, I dug into the vanilla client a bit more and realized that actually, while some things like entity types do have defaults, others that have been causing problems like sounds actually do not; if you launch Minecraft with the error log, you'll see that whatever thread is processing the packet is constantly printing NPE traces.

In light of this, I've decided to scrap the idea of default values and instead give library users more control over error handling. The latest PacketLib commit (which is now depended on by the latest MCProtocolLib commit) adds PacketErrorEvent, which has a method "setSuppress" that allows users to suppress the internal error handling and disconnect for any packet read/write errors that occur. Instead, apps are free to handle the error however they wish, perhaps by simply ignoring packets with bad values and printing a warning. I've also added some specific exception types for MagicValues related errors to make them easier to identify.
